app/app.go

```diff
@@ -450,3 +450,42 @@ func (a *App) Handle404(w http.ResponseWriter, r *http.Request) {
 
 	utils.RenderWebError(err, w, r)
 }
+
+// This function migrates the default built in roles from code/config to the database.
+func (a *App) DoAdvancedPermissionsMigration() {
```

Comment:
How do we prevent this from running simultaneously on multiple nodes of a clustered system? Or is it designed to be idempotent so that this isn't an issue?

Reply:
Good point. It just so happens that it *is* idempotent though, so I think it's fine.